fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Chirp.java

```diff
@@ -0,0 +1,93 @@
+package com.github.dedis.popstellar.model.objects;
+
+import androidx.annotation.NonNull;
+
+/** Class modeling a Chirp */
+public class Chirp {
+
+  private static final int MAX_CHIRP_CHARS = 300;
+
+  private String id;
+  private String channel;
+
+  private String sender;
+  private String text;
+  private long timestamp;
+  private int likes;
+  private String parentId;
+
+  public Chirp(String id) {
+    if (id == null) {
+      throw new IllegalArgumentException(" The id is null");
+    } else if (id.isEmpty()) {
+      throw new IllegalArgumentException(" The id of the Chirp is empty");
+    }
+    this.id = id;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(@NonNull String id) {
+    this.id = id;
```

Comment:
As you check that the id is not empty in the constructor, you might want to check it here too.